src/distributions/range.rs

```diff
@@ -454,11 +466,18 @@ macro_rules! range_float_impl {
                 // Generate a value in the range [1, 2)
                 let value1_2 = (rng.$next_u() >> $bits_to_discard)
                                .into_float_with_exponent(0);
-                // Doing multiply before addition allows some architectures to
-                // use a single instruction.
+                // We don't use `f64::mul_add`, because it is not available with
+                // `no_std`. Furthermore, it is slower for some targets (but
+                // faster for others).
                 value1_2 * self.scale + self.offset
             }
         }
+
+        impl From<::core::ops::Range<$ty>> for RangeFloat<$ty> {
```

Comment:
I don't see a point in keeping these when there also is the more generic implementation?

Reply:
Removing this would break `RangeFloat::from(range)`, wouldn't it? I would rather keep it, if `RangeFloat` stays public.

Reply:
`RangeFloat` is supposed to be an implementation detail, only there to make it easier to implement `Range` for user types. No one is supposed to use it directly.

Reply:
Then it should be private, right? If we make it private (or hide the docs) I would agree that the other impls should be removed.